### PR TITLE
Switch to pipenv for default installation

### PR DESCRIPTION
**Pull request by davidbgk**
I don't really know what I'm doing here :)

So please get the branch and try with your env.

During the process I hit https://github.com/kennethreitz/pipenv/issues/1012 so I manually added the dependency [as advised](https://github.com/kennethreitz/pipenv/issues/1012#issuecomment-342898222) but I don't know if it will works on different envs…

### CODE REVIEW

File: Pipfile.lock [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            "version": "==3.12"
+        },
+        "six": {
+            "hashes": [
+                "sha256:832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb",
+                "sha256:70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9"
+            ],
+            "version": "==1.11.0"
+        },
+        "tornado": {
```
Comment (yohanboniface) [inline Pipfile.lock]: Is it expected that dependencies of our dependencies ends here?

**Comment by yohanboniface** [comment]
I guess I hit it:

```
leonardo:~/C/p/roll (pipenv $=) pipenv install '-e .' --dev --three                                                                                                                        0ms
Creating a virtualenv for this project…
Using /usr/bin/python3 to create virtualenv…
⠋Running virtualenv with interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/ybon/.local/share/virtualenvs/roll-dOH39yhd/bin/python3
Also creating executable in /home/ybon/.local/share/virtualenvs/roll-dOH39yhd/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /home/ybon/.local/share/virtualenvs/roll-dOH39yhd
Installing -e .…
Traceback (most recent call last):
  File "/usr/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==8.3.2', 'console_scripts', 'pipenv')()
  File "/usr/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pipenv/cli.py", line 1880, in install
    converted = convert_deps_from_pip(package_name)
  File "/usr/lib/python3.6/site-packages/pipenv/utils.py", line 547, in convert_deps_from_pip
    os.path.isfile(req.name)) and not req.vcs:
  File "/usr/lib/python3.6/genericpath.py", line 30, in isfile
    st = os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```

**Comment by yohanboniface** [comment]
Still a lot of troubles with pipenv since then in all my tries.
I suggest to close here, and not to be too normative here anyway and let the user choose the best method for them.

**Comment by davidbgk** [comment]
Agreed, pipenv is not the silver bullet, and I tried hard :(

**Pull request closed without merging**